pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+# Copyright 2017 InovaDevelopment Inc. All Rights Reserved.
+
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+"""
+This package supports logging using the standard Python :mod:`py:logging`
+module. The logging support for multiple :class:`~py:logging.Logger` objects
+(named loggers):
+
+Currently there are two defined in this code:
+
+* 'pywbem.ops' for user-issued calls to pywbem CIM OperationsAPI functions, at
+  the info level. Internal calls to API functions are not logged.
+
+* 'pywbem.http' for http level information, at the info level.
+
+In addition, there may be loggers for pywbem modules with the module name, for
+situations like errors or warnings.
+
+All these loggers have a null-handler (see :class:`~py:logging.NullHandler`)
+and have no log formatter (see :class:`~py:logging.Formatter`).
+
+As a result, the loggers are silent by default. To turn on logging,
+add a log handler (see :meth:`~py:logging.Logger.addHandler`, and
+:mod:`py:logging.handlers` for the handlers included with Python) and set the
+log level (see :meth:`~py:logging.Logger.setLevel`, and :ref:`py:levels` for
+the defined levels).
+
+If you want to change the default log message format, use
+:meth:`~py:logging.Handler.setFormatter`. Its ``form`` parameter is a format
+string with %-style placeholders for the log record attributes (see Python
+section :ref:`py:logrecord-attributes`).
+
+Examples:
+
+* To output log records for one of the defined named loggers, either the
+  PywbemLogger.create_loggers or the Pywbem.create_logger can be used.
```

Comment:
`PywbemLogger.create_loggers` should be a reference to the method: `:meth:'PywbemLoggers.create_loggers'`. (and this would have caught the missing `s`).
Similar for `Pywbem.create_logger`.